lib/virtual.js

```diff
@@ -16,24 +53,64 @@ function VirtualModulesPlugin(compiler) {
 
 		compiler.inputFileSystem.purge = function() {
 			if (originalPurge) {
-				originalPurge.call(this, arguments);
+				originalPurge.apply(this, arguments);
 			}
 			if (this._virtualFiles) {
-				Object.keys(this._virtualFiles).forEach(
-					function(file) {
-						var data = this._virtualFiles[file];
-						setData(this._statStorage, file, [null, data.stats]);
-						setData(this._readFileStorage, file, [null, data.contents]);
-					}.bind(this)
-				);
+				Object.keys(this._virtualFiles).forEach(function(file) {
+					var data = this._virtualFiles[file];
+					this._writeVirtualFile(file, data.stats, data.contents);
+				}.bind(this));
 			}
 		};
 
 		compiler.inputFileSystem._writeVirtualFile = function(file, stats, contents) {
+			const statStorage = getStatStorage(this);
+			const fileStorage = getFileStorage(this);
+			const readDirStorage = getReadDirBackend(this);
 			this._virtualFiles = this._virtualFiles || {};
 			this._virtualFiles[file] = { stats: stats, contents: contents };
-			setData(this._statStorage, file, [null, stats]);
-			setData(this._readFileStorage, file, [null, contents]);
+			setData(statStorage, file, createWebpackData(stats));
+			setData(fileStorage, file, createWebpackData(contents));
+			var segments = file.split(/[\\/]/);
+			var count = segments.length - 1;
+			var minCount = segments[0] ? 1 : 0;
+			while (count > minCount) {
+				var dir = segments.slice(0, count).join(path.sep) || path.sep;
+				try {
+					compiler.inputFileSystem.readdirSync(dir);
+				} catch (e) {
+					var time = Date.now();
+					var dirStats = new VirtualStats({
+						dev: 8675309,
+						nlink: 0,
+						uid: 1000,
+						gid: 1000,
+						rdev: 0,
+						blksize: 4096,
+						ino: inode++,
+						mode: 16877,
+						size: stats.size,
+						blocks: Math.floor(stats.size / 4096),
+						atime: time,
```

Comment:
This should probably be rebased against `master` to include this recent change: https://github.com/sveltejs/svelte-loader/commit/277309765286aaff11c637540250e406e93ad6eb